specs/core/1_new_shards.md

```diff
@@ -0,0 +1,440 @@
+        proposer_index=proposer_index
+    )
+    
+    # Type 1: on-time attestations
+    if data.custody_bits != []:
+        # Correct slot
+        assert data.slot == state.slot
+        # Slot the attestation starts counting from
+        start_slot = state.shard_next_slots[shard]
+        # Correct data root count
+        offset_slots = [start_slot + x for x in SHARD_BLOCK_OFFSETS if start_slot + x < state.slot]
+        assert len(attestation.custody_bits) == len(offset_slots)
+        # Correct parent block root
+        assert data.beacon_block_root == get_block_root_at_slot(state, state.slot - 1)
+        # Apply
+        online_indices = get_online_indices(state)
+        if get_total_balance(state, online_indices.intersection(attesting_indices)) * 3 >= get_total_balance(state, online_indices) * 2:
+            # Check correct formatting of shard transition data
+            transition = block.shard_transitions[shard]
+            assert data.shard_transition_hash == hash_tree_root(transition)
+            assert len(transition.shard_data_roots) == len(transition.shard_states) == len(transition.shard_block_lengths) == len(offset_slots)
+            assert transition.start_slot == start_slot
+
+            # Verify correct calculation of gas prices and slots and chunk roots
+            prev_gasprice = state.shard_states[shard].gasprice
+            for i in range(len(offset_slots)):
+                shard_state, block_length, chunks = transition.shard_states[i], transition.shard_block_lengths[i], transition.shard_data_roots[i]
+                block_length = transition.shard
+                assert shard_state.gasprice == update_gasprice(prev_gasprice, block_length)
+                assert shard_state.slot == offset_slots[i]
+                assert len(chunks) == block_length // SHARD_BLOCK_CHUNK_SIZE
+                filled_roots = chunks + [EMPTY_CHUNK_ROOT] * (MAX_SHARD_BLOCK_CHUNKS - len(chunks))
+                assert shard_state.latest_block_hash == hash_tree_root(filled_roots)
+                prev_gasprice = shard_state.gasprice
+
+            # Save updated state
+            state.shard_states[shard] = transition.shard_states[-1]
+            state.shard_states[shard].slot = state.slot - 1
+
+            # Save success (for end-of-epoch rewarding)
+            pending_attestation.crosslink_success = True
+
+            # Apply proposer reward and cost
+            estimated_attester_reward = sum([get_base_reward(state, attester) for attester in attesting_indices])
+            increase_balance(state, proposer, estimated_attester_reward // PROPOSER_REWARD_COEFFICIENT)
+            for state, length in zip(transition.shard_states, transition.shard_block_lengths):
+                decrease_balance(state, proposer, state.gasprice * length)
+            
+    # Type 2: delayed attestations
+    else:
+        assert slot_to_epoch(data.slot) in (get_current_epoch(state), get_previous_epoch(state))
```

Comment:
Should this be within the last `SLOTS_PER_EPOCH` slots? That is how it is in phase 0 so there aren't asymmetries depending on which slot you are assigned to